asio/CMakeLists.txt

```diff
@@ -0,0 +1,240 @@
+# Minimum supported CMake version
+cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
+
+if(CONAN_COMPILER)
+
+  # ==========================================================
+  project(asio VERSION ${CONAN_PACKAGE_VERSION} LANGUAGES CXX)
+  # ==========================================================
+
+  include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+  conan_basic_setup(TARGETS)    # this set CONAN_TARGETS
```

Comment:
Use the `cmake_find_package*` generators. This is intrusive and has no place in the project CML.